src/tests/multihost/ipa/test_misc.py

```diff
@@ -303,3 +303,86 @@ def test_authentication_indicators(self, multihost):
                                                  ' |tail -10')
         ssh.close()
         assert 'indicators: 2' in search.stdout_text
+
+    def test_ssh_hash_knownhosts(self, multihost, reset_password,
+                                 setup_ipa_client, backupsssdconf):
+        """
+        :title: Current value of ssh_hash_known_hosts causes error in
+         the default configuration in FIPS mode.
+        :description: In SSSD the default value for ssh_hash_known_hosts
+         is set to true, It should be changed to false for consistency with
+         the OpenSSH setting that does not hashes host names by default
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2014249
+        :id: 1cee74c8-a0ad-44d4-8287-a32e3266de22
+        :customerscenario: false
+        :steps:
+            1. Stop SSSD
+            2. Configure SSSD with ssh having default value of
+               ssh_hash_known_hosts / ssh_hash_known_hosts = True /
+               ssh_hash_known_hosts = False
+            3. Remove /var/lib/sss/pubconf/known_hosts file
+            4. Start SSSD
+            5. Perform SSH using IPA user
+            6. Check if hostnames are hashed/unhashed in
+               /var/lib/sss/pubconf/known_hosts
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+            5. Should succeed
+            6. Hostnames should be hashed/unhashed as per the value of
+               ssh_hash_known_hosts
+        """
+        tools = sssdTools(multihost.client[0])
+        server_host = multihost.master[0].sys_hostname
+        rm_host_keys = "rm -rf /tmp/ssh_host0003_rsa*"
+        rm_known_hosts = "rm -rf /var/lib/sss/pubconf/known_hosts"
+        view_known_hosts = "cat /var/lib/sss/pubconf/known_hosts"
+        # adding host to IPA server
+        multihost.master[0].run_command(r"ssh-keygen -q -t rsa -N '' -C '' "
+                                        r"-f /tmp/ssh_host0003_rsa")
+        multihost.master[0].run_command("ipa host-mod %s --sshpubkey="
+                                        "\"$(cat /tmp/ssh_host0003_rsa.pub)\" "
+                                        "--updatedns"
+                                        % multihost.client[0].sys_hostname)
```

Comment:
I don't understand this.

1) You assign a key to `client[0].sys_hostname` but later test looks for `master[0].sys_hostname`. This doesn't make sense.
Probably this works because IPA master has own key always set? I don't know this. Could you check?

2) `ssh-keyscan` should be used to figure out key of a host, not a new random key generated and assigned in IPA db.
(but answer to 1) is more important)

Reply:
1. I'm not assigning a key to `client[0].sys_hostname`, here I'm running this command on the server that adds a key bound with the client host. It will be like `ipa host-mod client.ipa.test --sshpubkey="$(cat /tmp/ssh_host0003_rsa.pub)" --updatedns` where client.ipa.test is a client i.e `client[0].sys_hostname`

Reply:
You can call it "bound" instead of "assign" but that doesn't change a thing.

Later test looks for `server_host` (`if server_host in known_hosts.stdout_text ...`), not for a "client.ipa.test".

Reply:
yes because SSH is being done on client:
`multihost.client[0].run_command(cmd,stdin_text="Secret123",raiseonerr=False)`

Reply:
> yes because SSH is being done on client: `multihost.client[0].run_command(cmd,stdin_text="Secret123",raiseonerr=False)`

This triggers fetching know-hosts list from server to this client.
Then you search for server_host's key in this list.
But on the server you set a client's key.

Reply:
Working on this now, thanks for acknowledging.